utils/filesystem.go

```diff
@@ -26,3 +27,22 @@ func GetDiskUsageStats(path string) (dirSize, inodeCount uint64, err error) {
 
 	return dirSize, inodeCount, err
 }
+
+// IsDirectory tests whether the given path exists and is a directory.
+func IsDirectory(path string) error {
+	info, err := os.Stat(path)
+	if err != nil {
+		return err
+	}
+
+	if !info.Mode().IsDir() {
```

Comment:
If path is a symlink that points to a valid dir, does this fail and should it?

Reply:
Updated the docs to clarify that this follows symlinks. This is what the prior code did, so I kept it the same.